### PR TITLE
[ViPPET] Refactor benchmarking to support multi-pipeline stream execution

### PR DESCRIPTION
**Pull request by ktaube26**
### Description

This PR introduces significant enhancements and refactoring to support **multi-stream pipeline execution and benchmarking** in the Visual Pipeline and Platform Evaluation Tool (ViPPET). It replaces channel-based configurations with stream-based specifications, improves type safety, and adds robust test coverage.

---

### 1. **API Schema Updates**
- Replaced `PipelineParametersRun` and `PipelineParametersBenchmark` with:
  - `PipelineRunSpec` (name, version, streams)
  - `PipelineBenchmarkSpec` (name, version, stream_rate)
- Updated request models:
  - `PipelineRequestRun` → uses `pipeline_run_specs`
  - `PipelineRequestBenchmark` → uses `pipeline_specs`
- Modified `PipelineInstanceStatus` to include:
  - `total_streams`
  - `streams_per_pipeline` (typed as `PipelineRunSpec`)

### 2. **Pipeline Manager**
- Added `build_pipeline_command()` to dynamically construct GStreamer commands based on multiple pipelines and streams.
- Implemented singleton pattern for `PipelineManager`.

### 3. **Benchmark Logic**
- Introduced proportional stream allocation based on `stream_rate`.
- Removed redundant validation logic.
- Returns `BenchmarkResult` with `streams_per_pipeline` as `PipelineRunSpec`.

### 4. **Instance Manager**
- Refactored pipeline execution:
  - Builds pipeline command from `pipeline_run_specs`.
  - Updates instance metrics (`total_streams`, `streams_per_pipeline`).
- Enhanced error handling and cancellation logic.

### 5. **File & Structure Changes**
- Renamed `gstpipeline.py` → `pipelines/loader.py`.
- Removed obsolete methods and utilities (`download_file`, `replace_file_path`).
- Added `make_tee_names_unique()` for unique tee element naming in multi-stream pipelines.

### 6. **Tests**
- Comprehensive unit tests for:
  - PipelineManager command building.
  - InstanceManager lifecycle (start, stop, error handling).
  - Benchmark stream allocation and cancellation.
- Updated existing tests to align with new API models.
- Added edge case tests for invalid ratios and error scenarios.

### 7. **Pipeline Configurations**
- Updated predefined pipelines:
  - `simplevs` (license plate detection).
  - Split SmartNVR into `smartnvr-analytics` and `smartnvr-mediaonly`.

### 8. **Miscellaneous**
- Removed unused dependencies (`requests`).
- Improved logging for better traceability.

## **Breaking Changes**
- API contract changes for run and benchmark requests.
- Removal of legacy channel-based parameters.

## **Testing**
- All unit tests updated and passing.
- Added new tests for:
  - Stream allocation logic.
  - Pipeline command generation.
  - Instance lifecycle management.

### Checklist:

- [x] I agree to use the APACHE-2.0 license for my code changes.
- [x] I have not introduced any 3rd party components incompatible with APACHE-2.0. 
- [x] I have not included any company confidential information, trade secret, password or security token. 
- [x] I have performed a self-review of my code.

### CODE REVIEW

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/api/api_schemas.py [hunk 1]
```diff
@@ -110,17 +110,12 @@ class PipelineValidation(BaseModel):
...
+    pipeline_run_specs: list[PipelineRunSpec]
 
 
 class PipelineRequestBenchmark(BaseModel):
-    async_: Optional[bool] = Body(default=True, alias="async")
-    source: Source
-    parameters: PipelineParametersBenchmark
-    tags: Optional[Dict[str, str]]
+    fps_floor: int = 30
+    pipeline_specs: list[PipelineBenchmarkSpec]
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/api/api_schemas.py]: Maybe we could be consistent here?
Either use `pipeline_specs` in both classes, or use `pipeline_run_specs` and `pipeline_benchmark_specs`.

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py [hunk 2]
```diff
@@ -38,23 +40,94 @@ def __init__(self):
...
+            if i == len(pipeline_specs) - 1:
+                # Last pipeline gets all remaining streams to handle rounding
+                streams_per_pipeline.append(remaining_streams)
+            else:
+                # Calculate proportional streams and round
+                streams = round(total_streams * spec.stream_rate / 100)
+                streams_per_pipeline.append(streams)
+                remaining_streams -= streams
+
+        return streams_per_pipeline
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py]: In `BenchmarkResult`, the variable `streams_per_pipeline` is of type `List[PipelineRunSpec]`.
But in `_calculate_streams_per_pipeline`, the variable `streams_per_pipeline` is of type `List[int]`.

Maybe in the latter case we should rename it to `streams_per_pipeline_counts` so it doesn’t get confused?
The same would apply to the variable with the same name used in the `while True` loop in the `run()` method.

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py [hunk 3]
```diff
@@ -38,23 +40,94 @@ def __init__(self):
         self.runner = PipelineRunner()
         self.logger = logging.getLogger(__name__)
 
+    def _calculate_streams_per_pipeline(
+        self, pipeline_specs: list[PipelineBenchmarkSpec], total_streams: int
+    ) -> list[int]:
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py:46]: This method may be static.

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py [hunk 4]
```diff
@@ -119,17 +190,35 @@ def run(
                 n_streams = 1  # Prevent N from going below 1
 
         if best_config[0] > 0:
+            # Use best configuration found
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/benchmark.py]:
```suggestion
            # Use the best configuration found
```

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/managers/instance_manager.py [hunk 5]
```diff
@@ -119,8 +119,8 @@ def _build_instance_status(
             state=instance.state,
             total_fps=instance.total_fps,
             per_stream_fps=instance.per_stream_fps,
-            ai_streams=instance.ai_streams,
-            non_ai_streams=instance.non_ai_streams,
+            total_streams=instance.total_streams,
+            streams_per_pipeline=instance.streams_per_pipeline,
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/managers/instance_manager.py:123]: Can you change this:
```
            self.logger.info(
                f"Pipeline instance status for {instance_id}: {pipeline_instance_status}"
            )
```
to debug?
(line 144)

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/utils.py [hunk 6]
```diff
@@ -1,50 +1,85 @@
 import logging
-import os
 import re
-import requests
 
 logger = logging.getLogger("utils")
 
 TEMP_DIR = "/tmp/"
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/utils.py]: `TEMP_DIR` is not used anymore.

File: tools/visual-pipeline-and-platform-evaluation-tool/vippet/utils.py [hunk 7]
```diff
@@ -1,50 +1,85 @@
...
+    else:
+        logger.debug("No tee elements found in pipeline")
+        return pipeline_str
 
+    # Replace each tee name with a unique identifier
+    for original_name in tee_names:
+        # Extract digits from original name (e.g., "t0" -> "0", "t1" -> "1")
+        original_digits = "".join(filter(str.isdigit, original_name))
+        # Create unique name: t{pipeline_index}{stream_index}{original_digits}
+        new_name = f"t{pipeline_index}{stream_index}{original_digits}"
```
Comment (p-zak) [inline tools/visual-pipeline-and-platform-evaluation-tool/vippet/utils.py]: The current `make_tee_names_unique` implementation derives the new tee name only from the digits present in the original name. If we have multiple tees whose names do not contain digits (e.g. `tee name=a`, `tee name=b`) they can collapse to the same generated name and conflict.
To avoid potential collisions, it is safer to include a local index when generating the new name, so each tee in a pipeline is guaranteed to get a unique suffix even if their original names share the same digit pattern (or have no digits at all).

```suggestion
    # Replace each tee name with a unique identifier.
    # Using an index in addition to digits from the original name prevents
    # collisions when multiple tees have names without digits or share
    # the same digit pattern.
    for idx, original_name in enumerate(sorted(tee_names)):
        # Extract digits from original name (e.g., "t0" -> "0", "t1" -> "1")
        original_digits = "".join(filter(str.isdigit, original_name))
        # Create unique name: t{pipeline_index}{stream_index}{idx}{original_digits}
        new_name = f"t{pipeline_index}{stream_index}{idx}{original_digits}"
```
